packages/tailwindcss/src/variants.ts

```diff
@@ -900,9 +906,16 @@ export function createVariants(theme: Theme): Variants {
 
 function quoteAttributeValue(value: string) {
   if (value.includes('=')) {
+    // Do not allow invalid operators in attribute values
+    if (/[*$^|~]\s+=/.test(value)) {
```

Comment:
Should we try to always precompile regular expressions?

Reply:
That's something we could do, it's also a non-global one (`/g`) so it's stateless.

This only ever runs in `data-*` or `aria-*` variants when using arbitrary values that contain an `=` so I'm assuming this will be used very rarely. Especially because the `aria-foobar` already translates to `aria-[foobar="true"]`, but this doesn't go via this function since it's a bare value instead of an arbitrary value.

Reply:
I think this condition would break this variant:

```
data-[potato="^_="]
```

Feeling like the better overall solution to this problem should rely on `segment` — do we just need to segment the value on `=` and quote the last chunk if it isn't already quoted? Maybe we don't need to protect people from writing `data-[potato_^_=_foo]` and just let that be user error that generates junk.